### PR TITLE
🧑‍💻 Removing the Hard Coded Docs TOC – Tina Defined Docs TOC

### PR DESCRIPTION
**Pull request by isaaclombardssw**
Removed the old documents table of contents implementation. It was hard coded as a JSON file, and we want this to be easily modified in Tina.

Please see the changelog below...

- updated the prop names for the nav data from `oldNavDocs` to `navDocData` for clarity.
- removed an unused prop getting all docs in the props.
- added fields to the schema to represent the menu – a bit messy as new template classes are used to allow the menu to be nested up to 4 times and have items possible at every level. See `components/toc/toc-submenu.template.tsx` and `components/toc/toc-item.template.tsx`.

<details>

### General Contributing:

- [ ] Have you followed the guidelines in our [Contributing document](https://github.com/tinacms/tina.io/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### All New Content Submissions: (To be confirmed by reviewer)

- [ ] Title is short & specific
- [ ] Headers are logically ordered & consistent
- [ ] Purpose of document is explained in the first paragraph
- [ ] Procedures are tested and work
- [ ] Any technical concepts are explained or linked to
- [ ] Document follows structure from templates
- [ ] All links work
- [ ] The spelling and grammar checker has been run
- [ ] Graphics and images are clear and useful
- [ ] Any prerequisites and next steps are defined.

</details>

### CODE REVIEW

File: components/layout/setup-overview.tsx [hunk 1]
```diff
@@ -66,7 +66,7 @@ const OverviewTemplate = (props) => {
           // images: [openGraphImage(frontmatter.title, '| TinaCMS Docs')],
         }}
       />
-      <DocsLayout navItems={props.oldNavDocs.data}>
```
Comment (joshbermanssw) [inline components/layout/setup-overview.tsx:69]: Do we also remove all instances of oldNavDocs var?

**Review by joshbermanssw — APPROVED** [review]
other than the comment, i played around with it and looked good ✅ are there any options in incorporating some sort of feature in the screen where we create new doc pages to add it to the toc?

File: components/layout/setup-overview.tsx (hunk 1 above)
Comment (isaaclombardssw) [reply components/layout/setup-overview.tsx:69]: Yep

File: tina/schema.tsx [hunk 2]
```diff
@@ -716,7 +718,39 @@ export const schema = defineSchema({
       label: 'Whats new - TinaCloud',
       path: 'content/whats-new-tinacloud',
       format: 'mdx',
-      fields: WhatsNewFields,
+      fields: WhatsNewFields ,
+    },
+    {
+      name: 'docsTableOfContents',
+      label: 'Docs Table of Contents',
+      path: 'content/docs-toc',
```
Comment (bradystroud) [inline tina/schema.tsx:726]:
```suggestion
      name: 'docsTableOfContents',
      label: 'Docs - Table of Contents',
      path: 'content/docs-toc',
```

**Review by bradystroud — CHANGES_REQUESTED** [review]

**Comment by bradystroud** [comment]
Updating TinaCMS gave us this awesome new dropdown

image

**Comment by bradystroud** [comment]
oops - didn't mean to close

**Review by bradystroud — APPROVED** [review]
LGTM - this is awesome :)

**Pull request merged**